opencxl/cxl/component/cache_controller.py

```diff
@@ -0,0 +1,349 @@
+                min_idx = idx
+
+        return min_idx
+
+    def _cache_find_invalid_block(self, set: int) -> int:
+        for blk in range(self._cache_assoc_size):
+            if self._cache[set][blk].state == CacheState.CACHE_INVALID:
+                return blk
+
+        return -1
+
+    def _cache_find_valid_block(self, tag: int, set: int) -> int:
+        for blk in range(self._cache_assoc_size):
+            if (self._cache[set][blk].tag == tag) and (
+                self._cache[set][blk].state != CacheState.CACHE_INVALID
+            ):
+                return blk
+
+        return -1
+
+    def _cache_data_read(self, set: int, blk: int) -> int:
+        self._cache_priority_update(set, blk)
+
+        return self._cache[set][blk].data
+
+    def _cache_data_write(self, set: int, blk: int, data: int) -> None:
+        self._cache_priority_update(set, blk)
+        self._cache[set][blk].data = data
+
+    def _cache_rsp_state_loopup(self, packet: CacheResponse) -> CacheState:
+        if packet.status == CACHE_RESPONSE_STATUS.OK:
+            cache_state = CacheState.CACHE_EXCLUSIVE
+        elif packet.status == CACHE_RESPONSE_STATUS.RSP_S:
+            cache_state = CacheState.CACHE_SHARED
+        elif packet.status == CACHE_RESPONSE_STATUS.RSP_I:
+            cache_state = CacheState.CACHE_EXCLUSIVE
+        elif packet.status == CACHE_RESPONSE_STATUS.RSP_V:
+            cache_state = CacheState.CACHE_INVALID
+        return cache_state
+
+    async def _memory_load(self, address: int, size: int) -> CacheResponse:
+        # packet = CacheRequest(CACHE_REQUEST_TYPE.READ, address, size)
+        packet = CacheRequest(CACHE_REQUEST_TYPE.SNP_DATA, address, size)
+        await self._cache_to_coh_agent_fifo.request.put(packet)
+        packet = await self._cache_to_coh_agent_fifo.response.get()
+
+        return packet
+
+    async def _memory_store(self, address: int, size: int, value: int) -> None:
+        # packet = CacheRequest(CACHE_REQUEST_TYPE.WRITE, address, size, value)
+        packet = CacheRequest(CACHE_REQUEST_TYPE.WRITE_BACK, address, size, value)
+        await self._cache_to_coh_agent_fifo.request.put(packet)
+
+    # For request: coherency tasks from cache controller to coh module
+    async def _cache_to_coh_state_lookup(self, address: int) -> None:
+        packet = CacheRequest(CACHE_REQUEST_TYPE.SNP_INV, address)
+        await self._cache_to_coh_agent_fifo.request.put(packet)
+        packet = await self._cache_to_coh_agent_fifo.response.get()
+
+        assert packet.status == CACHE_RESPONSE_STATUS.RSP_I
+
+    # For response: coherency tasks from coh module to cache controller
+    async def _coh_to_cache_state_lookup(
+        self, type: CACHE_REQUEST_TYPE, addr: int
+    ) -> Tuple[int, int]:
+        data = 0
+        tag = self._cache_extract_tag(addr)
+        set = self._cache_extract_set(addr)
+
+        cache_blk = self._cache_find_valid_block(tag, set)
+
+        if cache_blk != -1:
```

Comment:
Could you return/check for "None" instead of -1 on these functions and checks?

Reply:
Sounds reasonable. I made fix and code recommitted.